### DISCUSSION TITLE
issue: getFieldState() does not update `isTouched`

### DISCUSSION THREAD

**Discussion by sambauers** [react-hook-form/react-hook-form#8142] [Q&A] [answered]
### Version Number

7.29.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-basic-forked-li3etz

### Steps to reproduce

1. Go to https://codesandbox.io/s/react-hook-form-basic-forked-li3etz
2. Click on first field in form `First Name`
3. Enter any valid value into field
4. Press `tab` key to advance to next field `Last Name`
5. Observe that `isTouched` in first field has not updated to `true`
6. Do not enter any value into required field `Last Name`
7. Press `tab` key to advance to next field `Email`
8. Observe that `isTouched` in first field has correctly updated to `true`
9. Observe that `isTouched` in second field has correctly updated to `true`

### Expected behaviour

I expect the `isTouched` value to update to `true` when the field is exited (`onBlur`) if a valid value is entered into the first field.

Relevant to this report is that I found a way to force the expected behaviour.

When `formState: { touchedFields }` is extracted from the `useForm` methods the `isTouched` values update as expected.

### What browsers are you seeing the problem on?

Safari

### Relevant log output

```shell
None
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
> formState is wrapped with a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to improve render performance and skip extra logic if specific state is not subscribed to. Therefore make sure you invoke or read it before a render in order to enable the state update.

missing `formState` touchedFields subscription.

```tsx
formState: { errors, touchedFields },
```

**Reply by sambauers** [reply]
OK, that approach is what I ended up discovering by accident. I see now in the docs the phrase and examples:

> formState will need to subscribed.

I'll take some time to create a PR to improve the description here (the code samples seem to make sense) as it was not clear to me what this all meant when I first saw it.

On a side note, stricter Typescript compilers and linters will complain about unused declarations, so I'll add a note about that too in any PR I create.

**Reply by bluebill1049** [reply]
thanks in advance 👍

**Reply by sambauers** [reply]
https://github.com/react-hook-form/documentation/pull/758

**Reply by sambauers** [reply]
I hope that's all correct.

**Reply by bluebill1049** [reply]
thanks mate.
